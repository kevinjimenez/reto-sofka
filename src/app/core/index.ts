export { spinnerInterceptor } from './interceptors/spinner.interceptor';
export { productsResolver } from './resolvers/products.resolver';
export { ProductsService } from './services/products.service';
export { SpinnerService } from './services/spinner.service';
