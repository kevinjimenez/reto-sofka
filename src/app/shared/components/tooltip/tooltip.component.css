.tooltip-content {
  position: absolute;
  background-color: #333;
  color: white;
  border-radius: 0.3rem;
  padding: 0.3rem;
  display: none;
  list-style-type: none;
  bottom: 100%;
  left: 50%;
  transform: translateX(-50%);
  z-index: 1000;
  box-shadow: 0px 4px 10px rgba(0, 0, 0, 0.1);
  font-size: 0.6rem;
}


/* Tooltip visible */
.tooltip-content.show {
  display: block;
}
