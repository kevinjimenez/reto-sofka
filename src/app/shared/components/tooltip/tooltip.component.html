<main class="tooltip-content" [ngClass]="{ show: isVisible() }">
	{{ text }}
</main>
