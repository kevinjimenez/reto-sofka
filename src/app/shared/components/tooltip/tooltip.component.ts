import { NgClass } from '@angular/common';
import { Component, Input, signal } from '@angular/core';

@Component({
	selector: 'app-tooltip',
	standalone: true,
	imports: [NgClass],
	templateUrl: './tooltip.component.html',
	styleUrl: './tooltip.component.css'
})
export class TooltipComponent {
	@Input() text: string = '';
	public isVisible = signal<boolean>(false);

	show() {
		this.isVisible.set(true);
	}

	hide() {
		this.isVisible.set(false);
	}
}
