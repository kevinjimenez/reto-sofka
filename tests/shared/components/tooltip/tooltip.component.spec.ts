import { ComponentFixture, TestBed } from '@angular/core/testing';
import { TooltipComponent } from '../../../../src/app/shared/components/tooltip/tooltip.component';

describe('TooltipComponent', () => {
	let component: TooltipComponent;
	let fixture: ComponentFixture<TooltipComponent>;
	let compiled: HTMLElement;

	beforeEach(async () => {
		await TestBed.configureTestingModule({
			imports: [TooltipComponent]
		}).compileComponents();
	});

	beforeEach(() => {
		fixture = TestBed.createComponent(TooltipComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
		compiled = fixture.nativeElement;
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('should match snapshot', () => {
		expect(compiled).toMatchSnapshot();
	});

	it('should show the tooltip when show() is called', () => {
		component.show();
		fixture.detectChanges();

		expect(component.isVisible()).toBeTruthy();
	});

	it('should hide the tooltip when hide() is called', () => {
		component.hide();
		fixture.detectChanges();

		expect(component.isVisible()).toBeFalsy();
	});
});
