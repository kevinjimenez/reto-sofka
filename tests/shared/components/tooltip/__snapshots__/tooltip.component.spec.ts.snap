// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TooltipComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <main
    class="tooltip-content"
  >
     

  </main>
</div>
`;
